### ISSUE TITLE
Large CSV shipping

### ISSUE THREAD

**Issue by P01s0nV** [Cyb3rWard0g/HELK#484]
Hello !

I'm working on a project aimed to automate forensic artefacts harvest during incident response for different clients.

My goal here is to write a Python script that will be executed on a Windows DC via GPO, gather a KAPE-like archive for each computer in the domain, and launch for each archive in parallel a shipper that will send the csv files to a remote HELK server, specifically to Logstash to be parsed. I also have performance constraints for the shipping.

In the POC for this project I was able to make it work using Filebeat, but it is not the tool best fitted here as I don't need to track the files but just send them once.

I was wondering if you knew what kind of shipper I could use here? The ideal solution would be a stand alone shipper with minimal installation required.

**Comment by devdua** [comment]
Try Vector - https://github.com/timberio/vector

**Comment by Cyb3rWard0g** [comment]
Niceeeee!!! I had not heard of Vector @devdua ! That looks promising. Would that answer your question @P01s0nV ? I believe you can also replace the HELK's pipeline and use your own for that specific project. We can then add it as an integration for a specific use case in the project if you do not mind. it would be great to have a use cases section.

**Comment by P01s0nV** [comment]
Yes, Vector looks quite nice! I tried parsing and shipping a csv file, but I had issues connecting to elasticsearch which actively refused it on either ports 9200 and 9300 and both http and https. I wonder if I have to activate something on elastic or if it just does not for any reason accept a connection from a pipeline outside of the machine it is on?

@Cyb3rWard0g yes a use cases section would be a great idea ! This is still a work in progress but if you are interested I can keep you updated on how this idea goes.

**Comment by P01s0nV** [comment]
Hey, so I found how to make it work.
So the issue was that port 9200 is not exposed, so just adding the ports under elasticsearch in the yaml file used by docker-compose for installation was enough.

I tried to send a file that took about 50 minutes to send (for merely 100Mb) using filebeat into logstash with a filter for parsing before sending the result into elasticsearch. Using Vector it only took 3 minutes.
One issue I had was that Vector would not stop after finishing a file, but it might be fixed in next week's release.
